sql-streaming-mqtt/src/main/scala/org/apache/bahir/sql/streaming/mqtt/HDFSMQTTSourceProvider.scala

```diff
@@ -0,0 +1,184 @@
+      None
+    } else {
+      Some(passwordString)
+    }
+
+    val connectionTimeout: Int = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.connectionTimeout,
+      MqttConnectOptions.CONNECTION_TIMEOUT_DEFAULT.toString
+    ).toInt
+    logInfo(s"Set connection timeout $connectionTimeout")
+
+    val keepAlive: Int = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.keetAliveInterval,
+      MqttConnectOptions.KEEP_ALIVE_INTERVAL_DEFAULT.toString
+    ).toInt
+    logInfo(s"Set keep alive interval $keepAlive")
+
+    val mqttVersion: Int = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.mqttVersion,
+      MqttConnectOptions.MQTT_VERSION_DEFAULT.toString
+    ).toInt
+    logInfo(s"Set mqtt version $mqttVersion")
+
+    val cleanSession: Boolean = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.cleanSession,
+      "true"
+    ).toBoolean
+    logInfo(s"Set clean session $cleanSession")
+
+    val qos: Int = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.qos,
+      "0"
+    ).toInt
+    logInfo(s"Set qos $qos")
+
+    val maxBatchMessageNum = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.maxBatchMessageNum, s"${Long.MaxValue}"
+    ).toLong
+    logInfo(s"Control max message number in one batch $maxBatchMessageNum")
+
+    val maxBatchMessageSize = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.maxBatchMessageSize, s"${Long.MaxValue}"
+    ).toLong
+    logInfo(s"Control max message content size in one batch $maxBatchMessageSize")
+
+    val maxRetryNumber = getParameterValue(
+      caseInsensitiveParameter,
+      MQTTConfig.maxRetryNumber, "3"
+    ).toInt
+    logInfo(s"Set max retry number $maxRetryNumber")
+
+    val mqttConnectOptions: MqttConnectOptions = new MqttConnectOptions()
+    mqttConnectOptions.setAutomaticReconnect(true)
+    mqttConnectOptions.setCleanSession(cleanSession)
+    mqttConnectOptions.setConnectionTimeout(connectionTimeout)
+    mqttConnectOptions.setKeepAliveInterval(keepAlive)
+    mqttConnectOptions.setMqttVersion(mqttVersion)
+    (username, password) match {
+      case (Some(u: String), Some(p: String)) =>
+        mqttConnectOptions.setUserName(u)
+        mqttConnectOptions.setPassword(p.toCharArray)
+      case _ =>
+    }
+
+    new HdfsBasedMQTTStreamSource(
+      sqlContext,
+      metadataPath,
+      brokerUrl,
+      topic,
+      clientId,
+      mqttConnectOptions,
+      qos,
+      maxBatchMessageNum,
+      maxBatchMessageSize,
+      maxRetryNumber)
+  }
+
+  override def shortName(): String = "hdfs-mqtt"
+
+  private def getParameterValue(
+      parameters: Map[String, String],
+      parameterName: String,
+      defaultValue: String): String = {
+
+    parameters.getOrElse(parameterName.toLowerCase(Locale.ROOT), defaultValue)
+  }
+}
+
+object HDFSMQTTSourceProvider {
+  val SCHEMA_DEFAULT = StructType(
```

Comment:
There is already a schema for MQTT messages in `MQTTStreamConstants` object. Why not reuse it?